### ISSUE TITLE
Clock not updating

### ISSUE THREAD

**Issue by AshenPaladin** [home-sweet-gnome/dash-to-panel#1229]
**Things to do first**
- Confirm that the problem persists when Dash-to-Panel is the only enabled extension. To do so, disable every other extension, then restart gnome-shell by running the `r` command from the prompt that appears when pressing Alt+F2 on an X.org session, or by logging out/in on a Wayland session.
Confirmed, I have seen this issue appear even with Dash to Panel being the only extension active.
- Look for Dash-to-Panel errors in your log. To do so, run the `journalctl /usr/bin/gnome-shell -f -o cat &` command and reproduce the problem.
I can't exactly reproduce the issue since it seems to happen randomly. The error log does show me a few errors, but I don't know if they are related.
```
Can't update stage views actor StWidget is on because it needs an allocation.
Ignoring excess values in shadow definition

```
- Search existing opened and closed issues to see if the problem has already been reported.
I haven't been able to find any similar variants of this bug opened or closed.

**Describe the bug**
At seemingly random intervals, the clock stops updating and remains stuck at a specific time. It only recovers if I click on the date and time widget. I am unable to voluntarily reproduce it, since I've been unable to find anything specific that might be causing it.

**Linux distribution and version**
Ubuntu 20.04

**GNOME Shell version**
Run `gnome-shell --version` from the command line to get this
GNOME Shell 3.38.1
**Dash-to-Panel version**
This can be seen in the Dash-to-Panel Settings in the About tab.
Dash-to-Panel Version 40
**Where was Dash-to-Panel installed from?**
The GNOME extensions website, GNOME Software store, your distribution's package manager, pre-installed with the distribution, directly from github, etc.
The GNOME extension website
**Screenshots / Video captures**
If applicable, add screenshots or a link to a video capture to help explain your problem.
I do not have any for the time being.
**Additional Info**

**Comment by StructByLightning** [comment]
I have the same issue, but for me it only started happening on Ubuntu 20.10. On 20.04 it worked fine. I'm using v40 of Dash to Panel.

**Reproduction:**
1. Turn on the _Panel Intellihide_ feature
2. Drag something to overlap with the panel so it hides
3. Wait a couple minutes
4. Move your mouse against the edge of the monitor to make the panel pop up (make sure you do this on an area where the clock isn't)
5. The clock should be out of date

Full settings (copied from the _export to file_ button in the about tab):
```
[/]
available-monitors=[2, 0, 1, 3]
hotkeys-overlay-combo='TEMPORARILY'
intellihide=true
intellihide-behaviour='ALL_WINDOWS'
intellihide-hide-from-windows=true
intellihide-pressure-threshold=200
intellihide-use-pressure=true
multi-monitors=false
panel-element-positions='{"0":[{"element":"showAppsButton","visible":true,"position":"stackedTL"},{"element":"activitiesButton","visible":false,"position":"stackedTL"},{"element":"leftBox","visible":true,"position":"stackedTL"},{"element":"taskbar","visible":true,"position":"stackedTL"},{"element":"centerBox","visible":true,"position":"stackedBR"},{"element":"rightBox","visible":true,"position":"stackedBR"},{"element":"dateMenu","visible":true,"position":"stackedBR"},{"element":"systemMenu","visible":true,"position":"stackedBR"},{"element":"desktopButton","visible":true,"position":"stackedBR"}],"1":[{"element":"showAppsButton","visible":true,"position":"stackedTL"},{"element":"activitiesButton","visible":false,"position":"stackedTL"},{"element":"leftBox","visible":true,"position":"stackedTL"},{"element":"taskbar","visible":true,"position":"stackedTL"},{"element":"centerBox","visible":true,"position":"stackedBR"},{"element":"rightBox","visible":true,"position":"stackedBR"},{"element":"dateMenu","visible":true,"position":"stackedBR"},{"element":"systemMenu","visible":true,"position":"stackedBR"},{"element":"desktopButton","visible":true,"position":"stackedBR"}],"2":[{"element":"showAppsButton","visible":true,"position":"stackedTL"},{"element":"activitiesButton","visible":false,"position":"stackedTL"},{"element":"leftBox","visible":true,"position":"stackedTL"},{"element":"taskbar","visible":true,"position":"stackedTL"},{"element":"centerBox","visible":true,"position":"stackedBR"},{"element":"rightBox","visible":true,"position":"stackedBR"},{"element":"dateMenu","visible":true,"position":"stackedBR"},{"element":"systemMenu","visible":true,"position":"stackedBR"},{"element":"desktopButton","visible":true,"position":"stackedBR"}],"3":[{"element":"showAppsButton","visible":true,"position":"stackedTL"},{"element":"activitiesButton","visible":false,"position":"stackedTL"},{"element":"leftBox","visible":true,"position":"stackedTL"},{"element":"taskbar","visible":true,"position":"stackedTL"},{"element":"centerBox","visible":true,"position":"stackedBR"},{"element":"rightBox","visible":true,"position":"stackedBR"},{"element":"dateMenu","visible":true,"position":"stackedBR"},{"element":"systemMenu","visible":true,"position":"stackedBR"},{"element":"desktopButton","visible":true,"position":"stackedBR"}]}'
panel-positions='{"0":"BOTTOM","1":"BOTTOM","2":"BOTTOM","3":"BOTTOM"}'
primary-monitor=3
show-apps-icon-file=''
```

**Comment by Coksnuss** [comment]
Can confirm, although I didn't try to disable my other running extensions to see if the bug remains. However, there never was a problem with this until a few days/weeks ago and I've not installed any new extensions beside the ones I am already using for quite some time now (Miniview, No topleft hot corner, GTile).

There is a small difference however to what the OP described:
> It only recovers if I click on the date and time widget.

This is **not** true for me. It already helps when I hover the date and time with the mouse cursor (-> it updates immediately)

`dconf dump /org/gnome/shell/extensions/dash-to-panel/`;
```
[/]
appicon-margin=6
appicon-padding=5
available-monitors=[0]
dot-color-1='#5294e2'
dot-color-2='#5294e2'
dot-color-3='#5294e2'
dot-color-4='#5294e2'
dot-color-unfocused-1='#5294e2'
dot-color-unfocused-2='#5294e2'
dot-color-unfocused-3='#5294e2'
dot-color-unfocused-4='#5294e2'
focus-highlight-color='#eeeeee'
hot-keys=true
hotkeys-overlay-combo='TEMPORARILY'
middle-click-action='LAUNCH'
panel-element-positions='{"0":[{"element":"showAppsButton","visible":true,"position":"stackedTL"},{"element":"activitiesButton","visible":false,"position":"stackedTL"},{"element":"leftBox","visible":true,"position":"stackedTL"},{"element":"taskbar","visible":true,"position":"stackedTL"},{"element":"centerBox","visible":true,"position":"stackedBR"},{"element":"rightBox","visible":true,"position":"stackedBR"},{"element":"systemMenu","visible":true,"position":"stackedBR"},{"element":"dateMenu","visible":true,"position":"stackedBR"},{"element":"desktopButton","visible":true,"position":"stackedBR"}],"1":[{"element":"showAppsButton","visible":true,"position":"stackedTL"},{"element":"activitiesButton","visible":false,"position":"stackedTL"},{"element":"leftBox","visible":true,"position":"stackedTL"},{"element":"taskbar","visible":true,"position":"stackedTL"},{"element":"centerBox","visible":true,"position":"stackedBR"},{"element":"rightBox","visible":true,"position":"stackedBR"},{"element":"systemMenu","visible":true,"position":"stackedBR"},{"element":"dateMenu","visible":true,"position":"stackedBR"},{"element":"desktopButton","visible":true,"position":"stackedBR"}]}'
panel-element-positions-monitors-sync=true
panel-size=32
shift-click-action='MINIMIZE'
shift-middle-click-action='LAUNCH'
show-apps-icon-file=''
show-showdesktop-delay=300
show-showdesktop-hover=true
show-window-previews=false
showdesktop-button-width=1
trans-use-custom-opacity=true
```

I am using Arch Linux with gnome-shell `1:3.38.1-1` and I am running dash-to-panel in its latest version, based on commit faf2ca26b9f832065ba56fa249828d87895a4eef. (Using https://aur.archlinux.org/packages/gnome-shell-extension-dash-to-panel-git/).

**Comment by ByteEnable** [comment]
DUP issue of:

https://github.com/home-sweet-gnome/dash-to-panel/issues/1206

Verified fixed in Gnome 3.38.2 on Fedora 33.

**Comment by Coksnuss** [comment]
I wasn't aware that this is a bug in gnome-shell. But it seems to have vanished for me as well since a couple of days (running 3.38.2 by now as well).

Issue should be closed then.
